kura/org.eclipse.kura.core.net/src/main/java/org/eclipse/kura/core/net/util/NetworkUtil.java

```diff
@@ -124,72 +107,127 @@ public static int convertIp4Address(String ipAddress) {
         return NetworkUtil.packIp4AddressBytes(addressBytes);
     }
 
-    public static int packIp4AddressBytes(short[] bytes) {
+    public static int packIp4AddressBytes(short[] bytes) throws KuraException {
+        if ((bytes == null) || (bytes.length != 4)) {
+            throw new KuraException(KuraErrorCode.INVALID_PARAMETER);
+        }
+        
         int val = 0;
-        for (int i = 3; i >= 0; i--) {
+        for (int i = 0; i < 4; i++) {
+            if ((bytes[i] < 0) || (bytes[i] > 255)) {
+                throw new KuraException(KuraErrorCode.INVALID_PARAMETER);
+            }
+            
             val = val << 8;
             val |= bytes[i];
         }
         return val;
     }
 
     public static short[] unpackIP4AddressInt(int address) {
-        return new short[] { (short) (address & 0xFF), (short) (address >> 8 & 0xFF), (short) (address >> 16 & 0xFF),
-                (short) (address >> 24 & 0xFF) };
+        short[] addressBytes = new short[4];
+        int value = address;
+        
+        for (int i = 3; i >= 0; i--) {
+            addressBytes[i] = (short)(value & 0xFF);
+            value = value >>> 8;
+        }
+        
+        return addressBytes;
     }
 
-    public static byte[] convertIP6Address(String fullFormIP6Address) {
+    public static byte[] convertIP6Address(String fullFormIP6Address) throws KuraException {
+        if (fullFormIP6Address == null) {
+            throw new KuraException(KuraErrorCode.INVALID_PARAMETER);
+        }
+
         byte[] retVal = new byte[16];
         String[] ip6Split = fullFormIP6Address.split(":");
-        for (int i = 0; i < 8; i++) {
-            ;
-            String octet = ip6Split[i];
-            StringBuffer sb = new StringBuffer();
 
-            while (sb.length() + octet.length() < 4) {
-                sb.append("0");
-            }
-            sb.append(octet);
+        if (ip6Split.length != 8) {
+            throw new KuraException(KuraErrorCode.INVALID_PARAMETER);
+        }
 
-            retVal[i * 2] = (byte) Short.parseShort(sb.toString().substring(0, 2), 16);
-            retVal[i * 2 + 1] = (byte) Short.parseShort(sb.toString().substring(2, 4), 16);
+        for (int i = 0; i < 8; i++) {
+            try {
+                String octet = ip6Split[i];
+                int value = Integer.parseInt(octet, 16);
+                
+                if ((value < 0) || (value > 0xFFFF))
+                {
+                    throw new KuraException(KuraErrorCode.INVALID_PARAMETER);
+                }
+                
+                int k = i * 2;
+                retVal[k] = (byte) ((value >>> 8) & 0xFF);
+                retVal[k + 1] = (byte) (value & 0xFF);
+            } catch (NumberFormatException e) {
+                throw new KuraException(KuraErrorCode.INVALID_PARAMETER);
+            }
```

Comment:
As noted above, IllegalArgumentException will be used and NFE is also added as root cause (fixed in also other places).